app/upgrades/v11/upgrades.go

```diff
@@ -58,3 +74,94 @@ func MigrateEscrowAccounts(ctx sdk.Context, ak authkeeper.AccountKeeper) {
 		ak.SetModuleAccount(ctx, acc)
 	}
 }
+
+// HandleMainnetUpgrade handles the logic for the reward distribution, it only commits to the db if successful
+func HandleRewardDistribution(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper, logger log.Logger) {
+	// use a cache context as a rollback mechanism in case
+	// the distrbution fails
+	cacheCtx, writeFn := ctx.CacheContext()
+	err := DistributeRewards(cacheCtx, bk, sk, dk)
+	if err != nil {
+		// log error instead of aborting the upgrade
+		logger.Error("failed to distribute rewards", "error", err.Error())
+	} else {
+		writeFn()
+	}
+}
+
+// DistributeRewards distributes the token allocations from the Olympus Mons incentivized testnet
+func DistributeRewards(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper, dk distributionkeeper.Keeper) error {
+	// TODO check the remaining rewards on each iteration to avoid sending more/less than supposed to (similar to v9.1 upgrade)
+	for _, currentDistribute := range Accounts {
+
+		// move rewards to the receiving account
+		receivingAccount := sdk.MustAccAddressFromBech32(currentDistribute[0])
+		receivingAmount, ok := sdk.NewIntFromString(currentDistribute[1])
+		if !ok {
+			return fmt.Errorf(
+				"reward distribution to address %s failed due to invalid parsing",
+				currentDistribute[0],
+			)
+		}
+		currentRewards := sdk.Coins{
+			sdk.NewCoin(types.BaseDenom, receivingAmount),
+		}
+		err := bk.SendCoins(ctx, sdk.MustAccAddressFromBech32(FundingAccount), receivingAccount, currentRewards)
+		if err != nil {
+			return fmt.Errorf(
+				"unable to send coins from fund account to participant account",
+			)
+		}
+
+		// stake from the receiving account to all validators equally
+		numValidators := len(Validators)
+		currentStakeAmount := (currentRewards.QuoInt(sdk.NewInt(int64(numValidators)))).AmountOf(types.BaseDenom)
+		remainderAmount := currentRewards.AmountOf(types.BaseDenom).Mod(sdk.NewInt(int64(numValidators)))
```

Comment:
`AmountOf` is unnecessary since you only have one coin in the slice. Same with the line above